src/compute/host_buffer.cpp

```diff
@@ -39,6 +39,40 @@ namespace xmipp4
 namespace compute
 {
 
+std::shared_ptr<device_buffer> 
+get_device_accessible_alias(const std::shared_ptr<host_buffer> &buffer) noexcept
+{
+    std::shared_ptr<device_buffer> result;
+
+    if (buffer)
+    {
+        auto *alias = buffer->get_device_accessible_alias();
+        if (alias)
+        {
+            result = std::shared_ptr<device_buffer>(buffer, alias);
+        }
+    }
+
+    return result;
+}
+
+std::shared_ptr<const device_buffer> 
+get_device_accessible_alias(const std::shared_ptr<const host_buffer> &buffer) noexcept
+{
+    std::shared_ptr<const device_buffer> result;
+
+    if (buffer)
+    {
+        const auto *alias = buffer->get_device_accessible_alias();
+        if (alias)
+        {
+            result = std::shared_ptr<const device_buffer>(buffer, alias);
+        }
+    }
+
+    return result;
+}
```

Comment:
I see the same code repeated over & over again in host_buffer & device buffer. It's the same thing but with 4 differences:
- `get_device_accesible_alias` returns `device_buffer`, while `get_host_accesible_alias` returns `host_buffer`.
- Then, inside of that differentiation, one of device returns const, the other without const.

Is there a way to dynamically specify the type being returned/worked with in the function and only define the behaviour once in a separate function that gets called 4 times (once in each device/host funcion)?

Reply:
Maybe I can work out something with templates, as in https://github.com/gigabit-clowns/xmipp4-compute-cuda/pull/20/commits/fe05b0e2735b21f50a0b22d661026ca0ff880d55

Reply:
I agree, would be nice and reduce code duplication

Reply:
I was able to reduce duplicity intra-type. However, reducing it across the types proves to be more challenging, due to the need of replacing a function call. There are ways to do this, but for this scale, I think their complexity is not worth